milabench/executors.py

```diff
@@ -0,0 +1,600 @@
+        """Execute all the commands and return the aggregated results
+        """
+        coro = []
+
+        for pack, argv, _kwargs in self.commands():
+            await pack.send(event="config", data=pack.config)
+            await pack.send(event="meta", data=machine_metadata())
+
+            pack.phase = "run"
+            fut = pack.execute(*argv, **{**_kwargs, **kwargs})
+
+            coro.append(fut)
+
+            if timeout:
+                delay = pack.config.get("max_duration", timeout_delay)
+                asyncio.create_task(force_terminate(pack, delay))
+
+        return await asyncio.gather(*coro)
+
+    def _argv(self, **kwargs) -> List:
+        del kwargs
+        return []
+
+
+# Leafs
+class CmdExecutor(Executor):
+    """Execute a command
+
+    Arguments:
+        pack: `BasePackage`'s instance from which `execute()` will be used to
+              perform the command
+        *cmd_argv: command line arguments list to execute
+        **kwargs: kwargs to be passed to the `pack.execute()`
+    """
+    def __init__(
+            self,
+            pack:pack.BasePackage,
+            *cmd_argv,
+            **kwargs
+    ) -> None:
+        if isinstance(pack, Executor):
+            raise ValueError(f"{self.__class__.__name__} does not accept nested"
+                             f" {Executor.__class__.__name__}")
+        super().__init__(
+            pack,
+            **kwargs
+        )
+        self.cmd_argv = cmd_argv
+
+    def _argv(self, **kwargs) -> List:
+        del kwargs
+        return [*self.cmd_argv]
+
+
+class PackExecutor(CmdExecutor):
+    """Execute a `Package`'s script. If not specified, the `Package`'s
+    main_script will be used
+
+    Arguments:
+        pack: `Package`'s instance from which `execute()` will be used to
+              perform the command
+        *script_argv: script's command line arguments list. If the first
+                      argument is a file that can be found, the file will be
+                      used instead of the `pack`'s main_script
+        **kwargs: kwargs to be passed to the `pack.execute()`
+    """
+    def __init__(
+            self,
+            pack:pack.Package,
+            *script_argv,
+            **kwargs
+    ) -> None:
+        script = script_argv[:1]
+        if script and XPath(script[0]).exists():
+            script = script[0]
+            script_argv = script_argv[1:]
+        else:
+            script = None
+
+        super().__init__(pack, *script_argv, **kwargs)
+
+        self.script = script
+
+    def _argv(self, **kwargs) -> List:
+        main = self.script or self.pack.main_script
+        if not XPath(main).is_absolute():
+            main = self.pack.dirs.code / main
+
+        if not main.exists():
+            raise FileNotFoundError(
+                f"Cannot run main script because it does not exist: {main}"
+            )
+
+        if main.is_dir():
+            main = ["-m", str(main)]
```

Comment:
The failure on the CI is due to
```suggestion
            main = ["-m", str(self.pack.main_script)]
```